### DISCUSSION TITLE
Heartbeat on Tasmota 12.2.0 ESP8266

### DISCUSSION THREAD

**Discussion by storfusker** [arendst/Tasmota#17110] [Q&A] [answered]
Anybody tried the heartbeat function with a LED connected to a gpio-pin on a ESP8266?

I already tried different gpio-pins, different high efficient LEDs in different colours and different serial resistors, but whatever I'm trying the heartbeat blink from the LED is very dim

It seems like the pulse out to the LED is far too short

Br
Steen

**Comment by arendst** [comment, marked as answer]
The heartbeat GPIO is used by secondary processors to signal the ESPxxx is still running. The pulse is very short and was never intended to be used as a "heartbeat led".

Look for other solution to have a blinking led.

**Reply by storfusker** [reply]
Thanks for the answer !

Ok that makes sense
Read it as a function to add a heartbeat LED, but I can see in the code that the pulse is only 50uSek what explains the very dim blink

Will look for another way of adding a heartbeat LED

And BTW...keep up the excellent job you're providing to the world

**Reply by sfromis** [reply]
Using Power blink along with BlinkCount and BlinkTime may be useful to you.....

**Comment by thomas-lentz** [comment]
Hi,
I hope I'm not doing something wrong by not creating a new discussion. My problem fits into this discussion perfectly:

I have a device (Air-gradient Pro), which uses a [TPL5010](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://www.ti.com/lit/gpn/TPL5010) as a Watchdog. The TP5010 needs a positive edge active signal with a pulselength of minimum 100ns.
I tried the TASMOTA Heartbeat signal, but it is not working. 
What is working is to use the MQTT Message LED signal, but since you can only assign this signal to one GPIO, my connection LED has no "job" anymore.
Of course I could hack the MQTT Message LED signal, but I think an Option for the Heartbeat length would be nice:

Option heartbeatlength
- n x 25us, where n is 0 to 255 (default 2, to not break the current functionality), allowing a range of 25 us to 6,375 ms, 
of course a non linear assignement with a value table would also be an option
- rising/falling Edge of the heartbeat signal is determined by "Heartbeat" and "Heartbeat_i"

Is this something which would be do-able?

Greetings Tom

**Reply by sfromis** [reply]
The mechanism is not really suitable for being made configurable length (like longer), as it blocks Tasmota while blinking. And is already a lot longer than 100ns.

**Reply by barbudor** [reply]
Why not using a PWM ?
You can set frequency and aspect ratio

If you want to prevent a slider, you can use a new led output, use led power to force it then ledpwm to set a duty cycle.
But I have idea if the actual timings

**Reply by sfromis** [reply]
ESP8266 `PWMFrequency` does not go lower than 40 Hz.

For relatively long "blinks", I already mentioned the Power blink feature as a possibility.
